### PR TITLE
BUG: GH10395 bug in DataFrame.interpolate with axis=1 and inplace=True

### LINKED ISSUE #10395: bug: interpolate() row-by-row doesn't work when setting inplace=True
```python

# pandas version 0.16.2
import numpy as np
import pandas as pd

np.random.seed(12345)
# simulate some data
df = pd.DataFrame(np.random.randn(250, 10), columns=np.arange(1, 11), index=pd.date_range('2015-01-01', periods=250, freq='B'))
# set 20% data to NaN
sel = np.random.choice([True, False], size=(250, 10), p=[0.2, 0.8])
df.values[sel] = np.nan

# row-wise interpolation doesn't work when setting inplace=True
df1 = df.copy()
df1.interpolate(method='linear', axis=1, inplace=True)

# inplace doesn't work
import pandas.util.testing as pdt
pdt.assert_frame_equal(df, df1)

# dropping inplace=True, it works
df2 = df.copy()
df2 = df2.interpolate(method='linear', axis=1)
pdt.assert_frame_equal(df, df2)

```

### PR DESCRIPTION
**Pull request by kawochen**
To close #10395.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -2925,45 +2925,50 @@ def interpolate(self, method='linear', axis=0, limit=None, inplace=False,
...
+            index=index,
+            values=_maybe_transposed_self,
+            limit=limit,
+            inplace=inplace,
+            downcast=downcast,
+            **kwargs
+        )
         if inplace:
             if axis == 1:
                 self._update_inplace(new_data)
```
Comment (jreback) [inline pandas/core/generic.py]: this does not need to be updated twice

Comment (jreback) [inline pandas/core/generic.py]: change to is more obvious what is going on. 
```
if axis == 1:
    new_data = self._constructor(new_data).T._data
self._update_inplace(new_data)
```

**Comment by kawochen** [comment]
@jreback  tests are green

**Comment by jreback** [comment]
@kawochen thanks! another nice fix!

**Pull request merged**